crates/nu-protocol/src/deprecation.rs

```diff
@@ -0,0 +1,142 @@
+use crate::{FromValue, ParseWarning, ShellError, Type, Value, ast::Call};
+
+// Make nu_protocol available in this namespace, consumers of this crate will
+// have this without such an export.
+// The `FromValue` derive macro fully qualifies paths to "nu_protocol".
+use crate::{self as nu_protocol, ReportMode, Span};
+
+#[derive(FromValue)]
+pub struct DeprecationEntry {
+    // might need to revisit this if we added additional DeprecationTypes
+    #[nu_value(rename = "flag", default)]
+    ty: DeprecationType,
+    #[nu_value(rename = "report")]
+    pub report_mode: ReportMode,
```

Comment:
Why is only this field `pub`?

Reply:
no idea, I just made the rest of them pub since it's a pub struct (which I think is probably reasonable?)

c2136a4b9